diff --git a/Android.mk b/Android.mk
index aedb3f9..957884a 100755
--- a/Android.mk
+++ b/Android.mk
@@ -233,14 +233,10 @@ ifeq ($(TW_FORCE_USE_BUSYBOX), true)
 else
     TW_USE_TOOLBOX := true
 endif
-ifeq ($(TW_EXCLUDE_MTP),)
-    LOCAL_SHARED_LIBRARIES += libtwrpmtp-ffs
 endif
-else
+
 ifeq ($(TW_EXCLUDE_MTP),)
-    LOCAL_CFLAGS += -DTW_HAS_LEGACY_MTP
-    LOCAL_SHARED_LIBRARIES += libtwrpmtp-legacy
-endif
+    LOCAL_SHARED_LIBRARIES += libtwrpmtp-ffs
 endif
 
 ifeq ($(BOARD_USES_RECOVERY_AS_BOOT), true)
@@ -897,11 +893,7 @@ ifeq ($(shell test $(PLATFORM_SDK_VERSION) -le 25; echo $$?),0)
 include $(commands_TWRP_local_path)/bootloader_message/Android.mk
 endif
 
-ifeq ($(shell test $(PLATFORM_SDK_VERSION) -ge 28; echo $$?),0)
-    include $(commands_TWRP_local_path)/mtp/ffs/Android.mk
-else
-    include $(commands_TWRP_local_path)/mtp/legacy/Android.mk
-endif
+include $(commands_TWRP_local_path)/mtp/ffs/Android.mk
 
 ifeq ($(wildcard system/core/uncrypt/Android.mk),)
     #include $(commands_TWRP_local_path)/uncrypt/Android.mk
diff --git a/mtp/ffs/MtpFfsCompatHandle.cpp b/mtp/ffs/MtpFfsCompatHandle.cpp
index 4027d60..c738edb 100644
--- a/mtp/ffs/MtpFfsCompatHandle.cpp
+++ b/mtp/ffs/MtpFfsCompatHandle.cpp
@@ -15,7 +15,7 @@
  */
 
 #include <android-base/logging.h>
-#include <android-base/properties.h>
+#include <cutils/properties.h>
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
diff --git a/mtp/ffs/MtpFfsHandle.cpp b/mtp/ffs/MtpFfsHandle.cpp
index 01b6f2e..c471462 100644
--- a/mtp/ffs/MtpFfsHandle.cpp
+++ b/mtp/ffs/MtpFfsHandle.cpp
@@ -15,7 +15,7 @@
  */
 
 #include <android-base/logging.h>
-#include <android-base/properties.h>
+#include <cutils/properties.h>
 #include <asyncio/AsyncIO.h>
 #include <dirent.h>
 #include <errno.h>
diff --git a/mtp/ffs/MtpServer.cpp b/mtp/ffs/MtpServer.cpp
index fa67024..67197a1 100755
--- a/mtp/ffs/MtpServer.cpp
+++ b/mtp/ffs/MtpServer.cpp
@@ -16,7 +16,7 @@
 
 #include <algorithm>
 #include <android-base/logging.h>
-#include <android-base/properties.h>
+#include <cutils/properties.h>
 #include <chrono>
 #include <dirent.h>
 #include <errno.h>
diff --git a/prebuilt/Android.mk b/prebuilt/Android.mk
index 17636ae..b1ae736 100644
--- a/prebuilt/Android.mk
+++ b/prebuilt/Android.mk
@@ -158,16 +158,10 @@ else
     TW_EXCLUDE_MTP := true
 endif
 
-ifeq ($(shell test $(PLATFORM_SDK_VERSION) -ge 28; echo $$?),0)
 ifneq ($(TW_EXCLUDE_MTP), true)
     RELINK_SOURCE_FILES += $(TARGET_OUT_SHARED_LIBRARIES)/libtwrpmtp-ffs.so
     RELINK_SOURCE_FILES += $(TARGET_OUT_SHARED_LIBRARIES)/libasyncio.so
 endif
-else
-ifneq ($(TW_EXCLUDE_MTP), true)
-    RELINK_SOURCE_FILES += $(TARGET_OUT_SHARED_LIBRARIES)/libtwrpmtp-legacy.so
-endif
-endif
 
 RELINK_SOURCE_FILES += $(TARGET_OUT_SHARED_LIBRARIES)/libext4_utils.so
 RELINK_SOURCE_FILES += $(TARGET_OUT_SHARED_LIBRARIES)/libaosprecovery.so
