From 11cf8ac4c1a69468aea564098c40c7e569cdb180 Mon Sep 17 00:00:00 2001
From: xunmod <116288337+xunmod@users.noreply.github.com>
Date: Mon, 20 Jan 2025 21:50:51 +0800
Subject: [PATCH] Fix Android 5.1 build

---
 partition.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/partition.cpp b/partition.cpp
index 3f25772..dee7fa1 100755
--- a/partition.cpp
+++ b/partition.cpp
@@ -1461,7 +1461,7 @@ bool TWPartition::Is_Mounted(void) {
         if (!Symlink_Mount_Point.empty()) {
             scan_mounted_volumes();
             const MountedVolume * sml = find_mounted_volume_by_mount_point(Symlink_Mount_Point.c_str());
-            if (sml != nullptr) {
+            if (sml != NULL) {
                 return true;
             }
         }
-- 
2.47.1

